src/components/FullscreenLoadingIndicator.js

```diff
@@ -25,18 +30,42 @@ const defaultProps = {
  * @param {Object} props
  * @returns {JSX.Element}
  */
-const FullScreenLoadingIndicator = (props) => {
-    if (!props.visible) {
-        return null;
+class FullScreenLoadingIndicator extends React.Component {
+    componentDidMount() {
+        if (!this.props.name || !this.props.timeout) {
+            return;
+        }
+
+        Log.info(`[LoadingIndicator] "${this.props.name}" became visible`);
+
+        this.timeoutId = setTimeout(
+            () => Log.alert(
+                `[LoadingIndicator] "${this.props.name}" is still visible after ${this.props.timeout} ms`,
```

Comment:
It's impossible to know this. But our alert logs must hit a threshold in order to actually create a ticket for us to investigate. I feel like with the low numbers of users we should use it here. There's a constant we can add and I'd change this to:

```js
Log.alert(`${CONST.ERROR.ENSURE_BUGBOT} [LoadingIndicator] Visible after timeout`,  details, false);
```

and put the timeout length and whatever other details into params